crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -133,6 +153,76 @@ def clusters_healthy?
       def compute_resources_available?
         Api::Crowbar.compute_resources_available?
       end
+
+      # Check Errors
+      # all of the below errors return an array with the following schema:
+      # [
+      #   {
+      #     code: String,
+      #     data: ... whatever data type ...,
+      #     help: String # "this is how you might fix the error"
+      #   },
+      #   ...
+      # ]
+      def sanity_check_errors
+        return [] if network_checks.empty?
+
+        [
+          {
+            code: :network_checks,
+            data: network_checks,
+            help: I18n.t("api.upgrade.prechecks.network_checks.help.default")
+          }
+        ]
+      end
+
+      def maintenance_updates_check_errors
+        return [] if maintenance_updates_status.empty?
+
+        [
+          {
+            code: :maintenance_updates_installed,
+            data: maintenance_updates_status[:errors],
+            help: I18n.t("api.upgrade.prechecks.maintenance_updates_check.help.default")
+          }
+        ]
+      end
+
+      def ceph_health_check_errors
+        return [] if ceph_healthy?
+
+        [
+          {
+            code: :ceph_health,
+            data: [], # TODO: implement ceph health check errors
```

Comment:
So, could we adapt `ceph_healthy?` in api/crowbar.rb to return such kind of errors in this PR?